### PR TITLE
Do not report CS1720 warning when member being accessed is a static member.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes issue #1845.

@VSadov, @agocke, @gafter, @jaredpar Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs [hunk 1]
```diff
@@ -4649,6 +4644,14 @@ private bool IsUsingAliasInScope(string name)
             }
         }
 
+        private static void WarnOnAccessOfOffDefault(CSharpSyntaxNode node, BoundExpression boundLeft, DiagnosticBag diagnostics)
+        {
+            if (boundLeft != null && boundLeft.Kind == BoundKind.DefaultOperator && boundLeft.ConstantValue == ConstantValue.Null)
+            {
```
Comment (VSadov) [inline src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs]: interesting .why not just check for constant null. i guess this is just to match old behavior.

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs]: Yes, this is an old code extracted into a method.

**Comment by gafter** [comment]
Is there no bottleneck to where the check could be moved after lookup?

**Comment by AlekseyTs** [comment]
I do not believe so. The check should happen after overload resolution, then we know the actual method used.

**Comment by gafter** [comment]
The bound node has a Symbol property that can be checked after calling the binding-continuation method. That should simplify the implementation considerably.

**Pull request merged**